### ISSUE TITLE
Change insertion markers to use JSON deserialization when duplicating blocks

### ISSUE THREAD

**Issue by BeksOmega** [google/blockly#7316]
### Check for duplicates

- [X] I have searched for similar issues before opening a new one.

### Problem

Insertion markers duplicate the block being dragged, and then tell the duplicate to be an insertion marker.

Currently the duplication duplicates (ha) a lot of code from the serialization system. We think this is because in the past the XML system didn't have a good way to just serialize and deserialize a single block (without children). But the JSON system does have this!

### Request

Change the insertion marker `createMarkerBlock` code over to use JSON serialization and deserialization.

### Additional context

Sparked by this discussion: https://groups.google.com/g/blockly/c/gcvynl88QLY/m/shFyX4x0BwAJ

**Comment by BeksOmega** [comment]
Sure go for it @varshneydevansh !

**Comment by varshneydevansh** [comment]
I found the discussion to be very intriguing and understood how we arrived at this issue.

To use the JSON serialization and deserialization the `toJSON` method is used to serialize the `sourceBlock` into a JSON object, which captures the relevant data of the block. The `fromJson` method is then called on the `result block` to deserialize the JSON and populate the block with the serialized data.

Is this is a suitable approach?
